src/install/dependency.zig

```diff
@@ -562,7 +562,9 @@ pub const Version = struct {
                                 }
                             }
 
+
                             if (strings.indexOfChar(url, '.')) |dot| {
+                                if (Repository.Hosts.has(url["git@".len..dot])) return .git;
```

Comment:
The start index can be larger than the end index with input like `ssh://.`, which is not valid in zig. Instead I would remove this line and add this before the check for `'.'`:
```zig
if (url.len > 4 and strings.eqlComptime(url[0.."git@".len], "git@")) {
    url = url["git@".len..];
}
```

Reply:
Done ✔️ Makes 100% sense and thank you